homeassistant/components/mazda/sensor.py

```diff
@@ -50,11 +44,9 @@ class MazdaSensorEntityDescription(
     unit: Callable[[UnitSystem], str | None] | None = None
 
 
-def _get_distance_unit(unit_system):
+def _get_distance_unit(unit_system: UnitSystem) -> str:
     """Return the distance unit for the given unit system."""
-    if unit_system.name == CONF_UNIT_SYSTEM_IMPERIAL:
-        return LENGTH_MILES
-    return LENGTH_KILOMETERS
+    return unit_system.length_unit
```

Comment:
The integration is currently using `unit_system.length` on the value functions, regardless of the unit system name.
This adjusts the `unit_of_measurement` function to match that behavior.